src/util-lua-packetlib.c

```diff
@@ -0,0 +1,265 @@
+/* Copyright (C) 2025 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * Packet API for Lua.
+ *
+ * local packet = require("suricata.packet")
+ */
+
+#include "suricata-common.h"
+
+#include "util-lua-packetlib.h"
+
+#include "app-layer-protos.h" /* Required by util-lua-common. */
+#include "util-lua-common.h"
+#include "util-lua.h"
+#include "util-debug.h"
+#include "util-print.h"
+
+/* key for p (packet) pointer */
+extern const char lua_ext_key_p[];
+static const char suricata_packet[] = "suricata:packet";
+
+struct LuaPacket {
+    Packet *p;
+};
+
+static int LuaPacketGC(lua_State *luastate)
+{
+    SCLogDebug("gc:start");
+    struct LuaPacket *s = (struct LuaPacket *)lua_touserdata(luastate, 1);
+    SCLogDebug("packet %p", s->p);
+    s->p = NULL;
+    SCLogDebug("gc:done");
+    return 0;
+}
+
+static int LuaPacketPayload(lua_State *luastate)
+{
+    struct LuaPacket *s = (struct LuaPacket *)lua_touserdata(luastate, 1);
+    if (s == NULL || s->p == NULL) {
+        LUA_ERROR("failed to get packet");
+    }
+
+    LuaPushStringBuffer(luastate, (const uint8_t *)s->p->payload, (size_t)s->p->payload_len);
+    return 1;
+}
+
+static int LuaPacketPacket(lua_State *luastate)
+{
+    struct LuaPacket *s = (struct LuaPacket *)lua_touserdata(luastate, 1);
+    if (s == NULL || s->p == NULL) {
+        LUA_ERROR("failed to get packet");
+    }
+
+    LuaPushStringBuffer(luastate, (const uint8_t *)GET_PKT_DATA(s->p), (size_t)GET_PKT_LEN(s->p));
+    return 1;
+}
+
+static int LuaPacketPcapCnt(lua_State *luastate)
+{
+    struct LuaPacket *s = (struct LuaPacket *)lua_touserdata(luastate, 1);
+    if (s == NULL || s->p == NULL) {
+        LUA_ERROR("failed to get packet");
+    }
+
+    lua_pushinteger(luastate, s->p->pcap_cnt);
+    return 1;
+}
+
+static int LuaPacketTimestring(lua_State *luastate)
+{
+    struct LuaPacket *s = (struct LuaPacket *)lua_touserdata(luastate, 1);
+    if (s == NULL || s->p == NULL) {
+        LUA_ERROR("failed to get packet");
+    }
+
+    char timebuf[64];
+    CreateTimeString(s->p->ts, timebuf, sizeof(timebuf));
```

Comment:
I think for a new API it would be really nice to generate a stanard timestring..  So instead of
```
YYYY/MM/DD-HH:mm:ss.ssssss
````
we should do:
```
YYYY-MM-DDTHH:mm:ss.ssssssZ
```

If needed, add a `_legacy` variant for the old one.  Or `timestamp_rfc3339` for this new one, and some other specifier for the legacy format?

Reply:
I'm open to suggestions for how to name the current output form I did (which is the common `CreateTimeString` method), but not interested at this point to explore crafting a new format. That can be a follow up ticket. It shouldn't block moving ahead with this I think.

Reply:
Leave as-is, document as "Suricata time string" or something along those lines with not about time zone.

`timestring_rfc3339` can come later.